### PR TITLE
Add Jetpack Cloud Server Connection Indicator Component

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Add `getRewindStateRequestStatus` selector
* Add new `ServerConnectionIndicator` Jetpack Cloud Component
* Add the component  to Jetpack Cloud `/settings/:site` route and refactor page in functional style

**Narrow ( not quite mobile ) screenshots**
narrow connected screenshot
narrow not-connected screenshot
**Wide Screenshots**
wide connected screenshot
wide not-connected screenshot

#### Testing instructions

1. Navigate to `/settings/:site` for a Jetpack site with some form of backup
2. Verify that after loading the page matches on of the above screenshots
3. [Using Redux DevTools dispatch the following action](https://github.com/reduxjs/redux-devtools)
```js
{
    type: 'REWIND_STATE_UPDATE',
    siteId: [your site id],
    data: {
        state: 'active'
    }
}
```
where `[your site id]` is your site id
4. Verify that the page matched the "Server Status: Connected" screenshot above
5. Using Redux DevTools dispatch the following action
```js
{
    type: 'REWIND_STATE_UPDATE',
    siteId: [your site id],
    data: {
        state: 'not-active'
    }
}
```
6. Verify that the page matched the "Server Status: Not connected" screenshot above
7. Verify that the Daily or Real-time on the page matches the plan for the site
8. Repeat steps 5-7 with a Jetpack site with the opposite type of plan

### CODE REVIEW

File: client/landing/jetpack-cloud/components/server-connection-indicator/placeholder.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+					viewBox="0 0 100 100"
+					xmlns="http://www.w3.org/2000/svg"
+					className="server-connection-indicator__placeholder-image "
+				>
+					<circle cx="50" cy="50" r="50" />
+				</svg>
+
+				<div className="server-connection-indicator__placeholder-info">
+					<h4 className="server-connection-indicator__placeholder-status">
+						{ translate( 'Server Status:' ) }
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/server-connection-indicator/placeholder.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **16** times:
`translate( 'Server Status' )` *ES Score: 10*

File: client/landing/jetpack-cloud/components/server-connection-indicator/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+					<div className="server-connection-indicator__message">
+						<p>{ message }</p>
+						{ ! isConnected && (
+							<ExternalLink
+								href="https://jetpack.com/support/adding-credentials-to-jetpack/"
+								target="_blank"
+								rel="noopener noreferrer"
+								icon={ true }
+							>
+								{ translate( 'Find your server credentials' ) }
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/components/server-connection-indicator/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Enter your server credentials' )` *ES Score: 6*

**Comment by cleacos** [comment]
For review, what is the status of this PR? could you rebase it to master since it has some conflict?

**Comment by cbauerman** [comment]
> For review, what is the status of this PR? could you rebase it to master since it has some conflict?

just going to chuck the PR for now, it won't be too hard to recreate.

**Pull request closed without merging**